### DISCUSSION TITLE
How do I specify the PAT to authenticate to a private nuget feed?

### DISCUSSION THREAD

**Discussion by efficienthacks** [dotnet/interactive#1340] [Q&A] [answered]
I know about this command
#i "nuget:[url]"

but how do I specify the PAT when it's a private nuget feed?

**Comment by efficienthacks** [comment, marked as answer]
So you have to use the dotnet command:
`dotnet nuget add source https://someServer/myTeam -n MyPrivateNugetPkg -u whatever -p adoPAT`

**Reply by efficienthacks** [reply]
I've noticed that this adds it to my user level nuget config ("C:\Users\username\AppData\Roaming\NuGet\NuGet.Config") - the advantage of doing it this way is that the PAT is encrypted in the nuget.config file.

**Comment by jonsequitur** [comment]
You can add a PAT inside a nuget.config as well. If you do this in your user-level nuget.config, it will work today, and you don't have to use `#i` at all. In the future we'll also support any nuget.config found in the same folder or a parent folder of the notebook file, using the same conventions used for project files.
